From f8bc6cd4749a26a6d6387bf63091c82281c884f4 Mon Sep 17 00:00:00 2001
From: Jianming Qiao <jianming.qiao@bp.renesas.com>
Date: Sun, 10 Nov 2019 17:06:23 +0000
Subject: [PATCH] Change test image set to grace_hopper.jpg

The name of the image in the Arm NN SDK Test Example is
hard coded. ARM does not provide any link for these test
images. Instead, we use standard grace_hopper image to
replace them.

Signed-off-by: Jianming Qiao <jianming.qiao@bp.renesas.com>
---
 tests/OnnxMobileNet-Armnn/OnnxMobileNet-Armnn.cpp  |  4 +--
 .../TfLiteMobilenetQuantized-Armnn.cpp             | 31 ++++-----------------
 tests/TfMobileNet-Armnn/TfMobileNet-Armnn.cpp      | 32 ++++------------------
 3 files changed, 13 insertions(+), 54 deletions(-)

diff --git a/tests/OnnxMobileNet-Armnn/OnnxMobileNet-Armnn.cpp b/tests/OnnxMobileNet-Armnn/OnnxMobileNet-Armnn.cpp
index 8786fea..321cc63 100644
--- a/tests/OnnxMobileNet-Armnn/OnnxMobileNet-Armnn.cpp
+++ b/tests/OnnxMobileNet-Armnn/OnnxMobileNet-Armnn.cpp
@@ -14,9 +14,7 @@ int main(int argc, char* argv[])
         // Coverity fix: The following code may throw an exception of type std::length_error.
         std::vector<ImageSet> imageSet =
         {
-            {"Dog.jpg", 208},
-            {"Cat.jpg", 281},
-            {"shark.jpg", 2},
+            {"grace_hopper.jpg", 652},
         };
 
         armnn::TensorShape inputTensorShape({ 1, 3, 224, 224 });
diff --git a/tests/TfLiteMobilenetQuantized-Armnn/TfLiteMobilenetQuantized-Armnn.cpp b/tests/TfLiteMobilenetQuantized-Armnn/TfLiteMobilenetQuantized-Armnn.cpp
index 1b411f9..c1c3635 100644
--- a/tests/TfLiteMobilenetQuantized-Armnn/TfLiteMobilenetQuantized-Armnn.cpp
+++ b/tests/TfLiteMobilenetQuantized-Armnn/TfLiteMobilenetQuantized-Armnn.cpp
@@ -30,31 +30,12 @@ std::vector<ImageSet> ParseDataset(const std::string& filename)
     }
     else
     {
-        // Use the default images
-        imageSet.push_back(ImageSet("Dog.jpg", 209));
-        // top five predictions in tensorflow:
-        // -----------------------------------
-        // 209:Labrador retriever 0.949995
-        // 160:Rhodesian ridgeback 0.0270182
-        // 208:golden retriever 0.0192866
-        // 853:tennis ball 0.000470382
-        // 239:Greater Swiss Mountain dog 0.000464451
-        imageSet.push_back(ImageSet("Cat.jpg", 283));
-        // top five predictions in tensorflow:
-        // -----------------------------------
-        // 283:tiger cat 0.579016
-        // 286:Egyptian cat 0.319676
-        // 282:tabby, tabby cat 0.0873346
-        // 288:lynx, catamount 0.011163
-        // 289:leopard, Panthera pardus 0.000856755
-        imageSet.push_back(ImageSet("shark.jpg", 3));
-        // top five predictions in tensorflow:
-        // -----------------------------------
-        // 3:great white shark, white shark, ... 0.996926
-        // 4:tiger shark, Galeocerdo cuvieri 0.00270528
-        // 149:killer whale, killer, orca, ... 0.000121848
-        // 395:sturgeon 7.78977e-05
-        // 5:hammerhead, hammerhead shark 6.44127e-055
+        // top five predictions in tensorflow lite:
+        // 653:military uniform 0.632812
+        // 458:bow tie, bow-tie, bowtie 0.0703125
+        // 452:bolo tie, bolo, bola tie, bola 0.015625
+        // 0:background 0
+        imageSet.push_back(ImageSet("grace_hopper.jpg", 653));
     };
     return imageSet;
 }
diff --git a/tests/TfMobileNet-Armnn/TfMobileNet-Armnn.cpp b/tests/TfMobileNet-Armnn/TfMobileNet-Armnn.cpp
index 6748b12..6b8370c 100644
--- a/tests/TfMobileNet-Armnn/TfMobileNet-Armnn.cpp
+++ b/tests/TfMobileNet-Armnn/TfMobileNet-Armnn.cpp
@@ -14,32 +14,12 @@ int main(int argc, char* argv[])
         // Coverity fix: The following code may throw an exception of type std::length_error.
         std::vector<ImageSet> imageSet =
         {
-            {"Dog.jpg", 209},
-            // Top five predictions in tensorflow:
-            // -----------------------------------
-            // 209:Labrador retriever 0.46392533
-            // 160:Rhodesian ridgeback 0.29911423
-            // 208:golden retriever 0.108059585
-            // 169:redbone 0.033753652
-            // 274:dingo, warrigal, warragal, ... 0.01232666
-
-            {"Cat.jpg", 283},
-            // Top five predictions in tensorflow:
-            // -----------------------------------
-            // 283:tiger cat 0.6508582
-            // 286:Egyptian cat 0.2604343
-            // 282:tabby, tabby cat 0.028786005
-            // 288:lynx, catamount 0.020673484
-            // 40:common iguana, iguana, ... 0.0080499435
-
-            {"shark.jpg", 3},
-            // Top five predictions in tensorflow:
-            // -----------------------------------
-            // 3:great white shark, white shark, ... 0.96672016
-            // 4:tiger shark, Galeocerdo cuvieri 0.028302953
-            // 149:killer whale, killer, orca, ... 0.0020228163
-            // 5:hammerhead, hammerhead shark 0.0017547971
-            // 150:dugong, Dugong dugon 0.0003968083
+            // 653:military uniform 0.779504
+            // 466:bulletproof vest 0.0485107
+            // 458:bow tie, bow-tie, bowtie 0.0130864
+            // 452:bolo tie, bolo, bola tie, bola 0.00530625
+            // 440:bearskin, busby, shako 0.00338392
+            {"grace_hopper.jpg", 653},
         };
 
         armnn::TensorShape inputTensorShape({ 1, 224, 224, 3  });
-- 
2.7.4

